firefox-ios/Storage/RustKeychain.swift

```diff
@@ -0,0 +1,68 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Common
+
+open class RustKeychain {
+    public private(set) var serviceName: String
+    public private(set) var accessGroup: String?
+
+    static var sharedClientAppContainerKeychain: RustKeychain {
+        let baseBundleIdentifier = AppInfo.baseBundleIdentifier
+
+        guard let accessGroupPrefix = Bundle.main.object(forInfoDictionaryKey: "MozDevelopmentTeam") as? String else {
+            return RustKeychain(serviceName: baseBundleIdentifier)
+        }
+        return RustKeychain(serviceName: baseBundleIdentifier,
+                            accessGroup: AppInfo.keychainAccessGroupWithPrefix(accessGroupPrefix))
+    }
+
+    public init(serviceName: String, accessGroup: String? = nil) {
+        self.serviceName = serviceName
+        self.accessGroup = accessGroup
+    }
+
+    struct KeychainError: Error {
+        let errorMessage: String
+    }
+
+    func getBaseKeychainQuery(key: String) -> [String: Any?] {
```

Comment:
`getBaseKeychainQuery` is derived from [setupKeychainQueryDictionary](https://github.com/mozilla/application-services/blob/c23b4bbc5a225af2a20457178ed5bf0d27927825/components/fxa-client/ios/FxAClient/MZKeychain/KeychainWrapper.swift#L408) minus properties that we didn't use in `RustLogins`.

Reply:
Do we also need `kSecAttrAccessible` here ? I see in [RustLogins](https://github.com/mozilla-mobile/firefox-ios/blob/098e52c5664ca121803127908e8ba284ffb70866/firefox-ios/Storage/Rust/RustLogins.swift#L240) we use `.afterFirstUnlock` which is not the default. According to [this](https://github.com/mozilla/application-services/blob/264e999c760880ddad426476f40cea5c31d17450/components/fxa-client/ios/FxAClient/MZKeychain/KeychainItemAccessibility.swift#L84-L87C10) default is `whenUnlocked`

Reply:
I set `kSecAttrAccessible` in the `setKeychainKey` function below. It's set to `kSecAttrAccessibleAfterFirstUnlock` which is equivalent to the value of the `MZKeychainItemAccessibility.afterFirstUnlock` enum variant. Since `setupKeychainQueryDictionary` returns the dictionary that all the `RustKeychain` functions rely on, and not all of the `MZKeychainWrapper` functions these functions are replacing set the`kSecAttrAccessible` property, I chose not to set it here.